migrations/2020-11-04-070016_sp_laporan/up.sql

```diff
@@ -0,0 +1,33 @@
+-- Counting Laporan for Numbering
+CREATE OR REPLACE FUNCTION sp_laporan_count_nomor(_id_jenis int, _id_satker int, _tanggal_laporan timestamp)
+RETURNS INTEGER AS $$
+BEGIN
+	RETURN (SELECT COUNT(*) AS count FROM laporan WHERE satker_id = _id_satker AND jenis_id = _id_jenis AND tanggal_laporan >= date_trunc('month', _tanggal_laporan));
```

Comment:
yang ini bukannnya perlu tahun ya?

Reply:
tahun nya sesuai timestamp _tanggal_laporan karena fungsi date_trunc nya cuman buat ngambil start date month nya

contoh:
SELECT date_trunc('month', now());

hasilnya:
2020-11-01 00:00:00+07

Reply:
ini kan untuk mengetahui posisi ketika ingin memasukan laporan, kan selalu di reset perbulannya,

nah kalau yang date_trunc itu kan dia di awal bulan, di bulan saat laporannya, saya bingung apakah akan terhitung dengan benar posisi si laporan ini,

Reply:
di latest commit query nya sudah seperti ini
tanggal_laporan >= date_trunc('month', _tanggal_laporan) AND tanggal_laporan < date_trunc('month', _tanggal_laporan + interval '1' month)

Reply:
```sql
the_month = date_trunc('month', tanggal_laporan) # 11
expected_month = date_trunc('month', _tanggal_laporan) #11

RETURN (SELECT COUNT(*) AS count FROM laporan WHERE satker_id = _id_satker AND jenis_id = _id_jenis AND the_month = expected_month;
```